src/ray/core_worker/task_event_buffer.cc

```diff
@@ -147,6 +141,15 @@ Status TaskEventBufferImpl::Start(bool auto_flush) {
     } else {
       RAY_LOG(INFO) << "Current task event io thread's nice = " << new_nice;
     }
```

Comment:
I think this is a misuse of `nice`.
* The value provided to `nice` is an increment. Search for `int nice (int increment) ` in the [glibc docs](https://www.gnu.org/software/libc/manual/pdf/libc.pdf)
* The return value of nice could be `-1` in a non-error case. This is because it returns the nice value. Apparently one needs to check `errno` as well as the return value to determine if there was an error. ([syscall docs](https://man7.org/linux/man-pages/man2/nice.2.html#RETURN_VALUE))

Reply:
Ah good call!